src/core/public/application/application_service.test.ts

```diff
@@ -413,6 +415,36 @@ describe('#setup()', () => {
         })
       );
     });
+
+    it('allows to update the basePath', async () => {
+      const setup = service.setup(setupDeps);
+
+      const pluginId = Symbol('plugin');
+      setup.register(pluginId, createApp({ id: 'app1' }));
+
+      const updater = new BehaviorSubject<AppUpdater>(app => ({}));
+      setup.registerAppUpdater(updater);
+
+      const start = await service.start(startDeps);
+      await start.navigateToApp('app1');
+      expect(MockHistory.push).toHaveBeenCalledWith('/app/app1', undefined);
+      MockHistory.push.mockClear();
+
+      updater.next(app => ({ defaultPath: 'default-path' }));
```

Comment:
from this example I'd expect it to be called as `defaultSubpath` / `internalPath`, etc.

Reply:
I think the `default` part is important as the prop is used as a default in `navigateToApp`.
However in `navigateToApp`, the app path option is called `path`, not `subPath`, so I choose `defaultPath`.

But I'm fine with `defaultSubPath` if we think this is more explicit.